src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -3594,27 +3725,35 @@ exprt c_typecheck_baset::do_special_functions(
     // do not typecheck the argument - it is never evaluated, and thus side
```

Comment:
The comment would appear to be from the time when the type checker has taken care of side effects. The argument is type checked.

Reply:
Removed.